### DISCUSSION TITLE
TypeDefinition 'Deposit' not found in parsed GraphQL schema.

### DISCUSSION THREAD

**Discussion by chlenc** [FuelLabs/fuel-indexer#1237] [Q&A] [answered]
#  TypeDefinition 'Deposit' not found in parsed GraphQL schema during the building of indexer

Hello there
I'm trying to setup the fuel indexer to [this contract](https://github.com/compolabs/spark/blob/beta-3/contracts/orderbook/src/main.sw)

In the contract, I have a[ few events](https://github.com/compolabs/spark/blob/beta-3/contracts/orderbook/src/structs.sw#L53) like that:
```
pub struct DepositEvent{
    timestamp: u64,
    address: Address,
    amount: u64,
}
```

I have defined some [handlers in `src/lib.rs`](https://github.com/compolabs/spark/blob/beta-3/spark-indexer/src/lib.rs#L11)
```
    fn handle_deposit_event(data: DepositEvent) {
        info!("💰 Deposit event \n{:#?}", data);
        let entry = Deposit {
            id: data.timestamp,
            timestamp: data.timestamp,
            address: data.address,
            amount: data.amount,
        };
        deposit.save();
    }
```

And ofc I have GraphQL types in [`schema/spark_indexer.schema.graphql` ](https://github.com/compolabs/spark/blob/beta-3/spark-indexer/schema/spark_indexer.schema.graphql#L28)
```
type Deposit {
    id: ID!
    timestamp: UInt8!
    address: Address!
    amount: UInt8!
}
```

But when I'm trying to build that I get an error 
<img width="617" alt="image" src="https://github.com/FuelLabs/fuel-indexer/assets/35231749/cdf8333d-95ff-42f4-97a3-da2a00367894">

## toolchain
```
Default host: aarch64-apple-darwin
fuelup home: /Users/alexey/.fuelup

installed toolchains
--------------------
beta-3-aarch64-apple-darwin
latest-aarch64-apple-darwin (default)
hotfix
my-custom-toolchain

active toolchain
-----------------
latest-aarch64-apple-darwin (default)
  forc : 0.42.1
    - forc-client
      - forc-deploy : 0.42.1
      - forc-run : 0.42.1
    - forc-doc : 0.42.1
    - forc-explore : 0.28.1
    - forc-fmt : 0.42.1
    - forc-index : 0.19.3
    - forc-lsp : 0.42.1
    - forc-tx : 0.42.1
    - forc-wallet : 0.2.4
  fuel-core : 0.18.3
  fuel-indexer : 0.19.3

fuels versions
---------------
forc : 0.43
```

## How to reproduce the bug
```
 git clone https://github.com/compolabs/spark.git
 cd spark
 git checkout beta-3 
 cd spark-indexer/  
 forc index build
```

**Comment by ra0x3** [comment, marked as answer]
@chlenc We made a breaking change in 0.19.0 to use _proper/idiomatic_ GraphQL directives

The way things used to work:
```graphql
type Deposit {
    id: ID!
    timestamp: UInt8!
    address: Address!
    amount: UInt8!
}
```

The way things work now (note that type definitions are decorated with the `@entity` directive 
```graphql
type Deposit @entity {
    id: ID!
    timestamp: UInt8!
    address: Address!
    amount: UInt8!
}
```

Let me know if this helps :)

**Comment by chlenc** [comment]
> ```graphql
>  @entity
> ```

Thanks for a super fast reply ser
Let me try 🫡

**Comment by chlenc** [comment]
> @chlenc We made a breaking change in 0.19.0 to use _proper/idiomatic_ GraphQL directives
> 
> The way things used to work:
> 
> ```graphql
> type Deposit {
>     id: ID!
>     timestamp: UInt8!
>     address: Address!
>     amount: UInt8!
> }
> ```
> 
> The way things work now (note that type definitions are decorated with the `@entity` directive
> 
> ```graphql
> type Deposit @entity {
>     id: ID!
>     timestamp: UInt8!
>     address: Address!
>     amount: UInt8!
> }
> ```
> 
> Let me know if this helps :)

That works, thanks a lot ❤️

**Reply by ra0x3** [reply]
Awesome @chlenc, do you mind marking that answer as the correct answer? I've converted this issue into a discussion, in case someone else has the same question :)
